hws/src/dr_ste.py

```diff
@@ -247,6 +247,16 @@ def dump_str(self, verbosity, prefix='STE '):
         _str = prefix + self.data.get("id") + ':\n'
         return _str
 
+    def ste_index_str(self, ste_index):
```

Comment:
why this is needed, is it for the resource sharing case?
will in that case all the info will be printed in context, table and matcher dump lines, and the tool will know how to handle the resources from different GVMI's

Reply:
This is required so that whenever an STE index/address is printed, the GVMI number is included if the STE is outside the current GVMI. This avoids confusion between a jump to "current GVMI, STE ID N" vs. "some other GVMI, STE ID N" where the same STE index/address just happens to exist in both places. When the STE is in another GVMI, it explicitly prints that GVMI so the user is aware they need to go look for it in some other "non-default" place, rather than in the current GVMI and then being confused when the STE doesn't exist or has some content that doesn't make sense.

Overall, every STE's location is _always_ a tuple of (GVMI, STE index/address), and never _just_ the plain STE index/address. Thus, the tool must print that entire tuple so that the dump accurately displays what the actions/... do. Sure, we can _default_ to not printing the GVMI when it happens to match the current/default GVMI, but when it doesn't this information is critical to correctly interpreting the operation of the actions, and thus must be printed.